docs/src/main/asciidoc/spring-security.adoc

```diff
@@ -0,0 +1,252 @@
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `spring-security-quickstart` {quickstarts-tree-url}/spring-security-quickstart[directory].
+
+== Creating the Maven project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,shell,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=spring-security-quickstart \
+    -DclassName="org.acme.spring.web.GreetingController" \
+    -Dpath="/greeting" \
+    -Dextensions="spring-web, spring-security, quarkus-elytron-security-properties-file"
+cd spring-security-quickstart
+----
+
+This command generates a Maven project with a REST endpoint and imports the `spring-web`, `spring-security` and `security-properties` extensions.
+
+For more information about `security-properties` you can check the guide of link:security-properties[quarkus-elytron-security-properties-file] extension.
+
+== GreetingController
+
+The Quarkus maven plugin automatically generated a controller with the Spring Web annotations to define our REST endpoint (instead of the JAX-RS ones used by default)
+The `src/main/java/org/acme/spring/web/GreetingController.java` file looks as follows:
+
+[source,java]
+----
+package org.acme.spring.web;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.PathVariable;
+
+@RestController
+@RequestMapping("/greeting")
+public class GreetingController {
+
+    @GetMapping
+    public String hello() {
+        return "hello";
+    }
+}
+----
+
+== GreetingControllerTest
+
+Note that a test for the controller has been created as well:
+
+[source, java]
+----
+package org.acme.spring.web;
+
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Test;
+
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+
+@QuarkusTest
+public class GreetingControllerTest {
+
+    @Test
+    public void testHelloEndpoint() {
+        given()
+          .when().get("/greeting")
+          .then()
+             .statusCode(200)
+             .body(is("hello"));
+    }
+
+}
+----
+
+== Package and run the application
+
+Run the application with: `./mvnw compile quarkus:dev`.
+Open your browser to http://localhost:8080/greeting.
+
+The result should be: `{"message": "hello"}`.
+
+== Modify the controller to secure the `hello` method
+
+Since we are going to be using Spring Security annotations to define the authorizations to execute the `hello` method, you can add the Spring `@Secured` annotation allowing secure the resources by restricting which roles are able to execute it. The controller above will look as follows:
```

Comment:
```suggestion
Since we are going to be using Spring Security annotations to define the authorizations to execute the `hello` method, add the Spring `@Secured` annotation to secure the resources by restricting which roles are able to execute it. The controller above will look as follows:
```

Reply:
I have rephrased this as @geoand suggested [here](https://github.com/quarkusio/quarkus/pull/5792#discussion_r351217720)